packages/emotion/src/styleUtils/ThemeablePropValues.ts

```diff
@@ -83,7 +83,39 @@ const ThemeablePropValues = {
     xLarge: 'x-large',
     xxLarge: 'xx-large'
   }
-}
+} as const
+
+type SpacingKeys = keyof typeof ThemeablePropValues.SPACING
+export type SpacingValues = typeof ThemeablePropValues.SPACING[SpacingKeys]
+export type Spacing =
+  | `${SpacingValues}`
+  | `${SpacingValues} ${SpacingValues}`
+  | `${SpacingValues} ${SpacingValues} ${SpacingValues}`
+  | `${SpacingValues} ${SpacingValues} ${SpacingValues} ${SpacingValues}`
```

Comment:
This is a relatively new TS feature called [`template literal types`](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html), which enable us to have proper types and IDE help for cases like:
```tsx
<View margin="xx-large" ...>
<View margin="xx-large auto" ...>
<View margin="xx-large auto small" ...>
```
and so on...

So effectively the users IDE will help them with all of the possible values which `Spacing` can take.